src/plugins/gs-plugin-external-apps.c

```diff
@@ -43,8 +44,15 @@
 
 #define TMP_ASSSETS_PREFIX "gs-external-apps"
 
+enum GsPluginExternalType {
+	GS_PLUGIN_EXTERNAL_TYPE_DEB = 1,
+	GS_PLUGIN_EXTERNAL_TYPE_TAR,
+	GS_PLUGIN_EXTERNAL_TYPE_UNKNOWN,
+};
+
 struct GsPluginData {
 	GsFlatpak	*flatpak;
+	enum GsPluginExternalType type;
```

Comment:
I'd prefer that you typedef'ed this type in the previous enum declaration. So we'd skip using *enum* here.